pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -435,13 +438,24 @@ func filterUnmountedVolumes(mountedVolumes sets.String, expectedVolumes []string
 // getExpectedVolumes returns a list of volumes that must be mounted in order to
 // consider the volume setup step for this pod satisfied.
 func getExpectedVolumes(pod *v1.Pod) []string {
-	expectedVolumes := []string{}
+	expectedVolumes := sets.NewString()
 
-	for _, podVolume := range pod.Spec.Volumes {
-		expectedVolumes = append(expectedVolumes, podVolume.Name)
+	for _, container := range pod.Spec.Containers {
```

Comment:
what about initContainers?

Reply:
Good catch!

I copied the code from VolumeManager and it observes only `.containers`, not `.initContainers`. Filed #81343. I think the best would be to fix `makeVolumeMap` in #81343 first (to have a small patch), backport it everywhere and then move it to `pkg/volume/utils` that will be used also here.

Reply:
Wrote too early, the issue is reproducible only in this PR.

I extracted `GetPodVolumeNames` and used it consistently in VolumeManager and DSWP.